hardware/ASyncTCP.h

```diff
@@ -8,61 +8,81 @@
 #include <boost/smart_ptr/shared_ptr.hpp>  // for shared_ptr
 #include <exception>                       // for exception
 
+#define ASYNCTCP_THREAD_NAME "ASyncTCP"
+
 namespace boost { namespace system { class error_code; } }
 
 class ASyncTCP
 {
-public:
+protected:
 	ASyncTCP();
 	virtual ~ASyncTCP(void);
+	// Async connect and start async read from the socket, data is delivered in OnData()
+	// No action if mIsConnected = true. If the connection is lost, it will automatically be setup again
+	void connect(const std::string &hostname, unsigned short port);
```

Comment:
setting mIsConnected could be part of the connect call (maybe default false)

Reply:
That would be assuming things. mIsConnected should and is set by `handle_connect()` when the peer responds to that request.

Reply:
Sorry, i meant the reconnect 'mDoReconnect' and the retry delay
Actually only the retry_delay, we do not need 'mDoReconnect'. When the retry_delay = 0, there won't be a reconnect